app/controllers/users_controller.rb

```diff
@@ -0,0 +1,59 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!
+
+  def settings
```

Comment:
Could you please move these logic from your controller to some service object or interactors? 
https://github.com/collectiveidea/interactor
https://www.toptal.com/ruby-on-rails/rails-service-objects-tutorial

It would be better when your controllers will have only REST actions (index, show, new, create, edit, update, delete)